### PR TITLE
Update data migration

### PR DESCRIPTION
**Pull request by andreievg**
I’ve checked, migration code works, rc1 to rc10

### CODE REVIEW

File: src/utilities/compareVersions.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    error.message = `error verifying versions, versionOne: ${versionOne}, versionTwo ${versionTwo}`;
+    bugsnagClient.notify(error);
+  }
+  return 0;
+}
+// versionToInteger logic is the same as ./android/app/build.gradle
+function versionToInteger(version) {
+  const [majorMinorPatch, provisional] = version.split('-');
+  const [major, minor, patch] = majorMinorPatch.split('.');
+  return Number(major) * 10000000 +
```
Comment (Chris-Petty) [inline src/utilities/compareVersions.js]: I think casting using `Number` is my favourite, but out of curiosity I went searching for why to use it over parseInt or parseFloat. Found a brilliant table on the matter!

https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which/17106702#17106702

Some people debate the performance of each method, but from what I've read most discussion is naive as it's platform specific.

File: src/utilities/compareVersions.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+         Number(minor) * 100000 +
+         Number(patch) * 100 +
+         Number(!provisional ? '99' : provisional.match(/[0-9]+/)[0]);
+         // Regex /[0-9]+/ matches only numbers and at least on digit number
```
Comment (Chris-Petty) [inline src/utilities/compareVersions.js]: This sentence might not be worded correctly?

File: src/utilities/compareVersions.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export function test() {
```
Comment (Chris-Petty) [inline src/utilities/compareVersions.js]: Would be better in a Jest test but don't bother. We don't have any level of automated tests sadly.

File: src/utilities/compareVersions.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    const testResult = compareVersions(versionOne, versionTwo);
+    const testMessage = testResult === expectedResult ? 'pass' : 'FAIL';
+    console.log(`checking ${versionOne} vs ${versionTwo}, status: ${testMessage}`);
+    console.log(`expected result: ${expectedResult}, actual result: ${testResult}`);
+  };
+
+  testHelper('1.0.0', '1.0.1', -1);
+  testHelper('1.0.0', '1.1.0', -1);
+  testHelper('1.0.0', '2.0.0', -1);
+  testHelper('2.0.0', '2.0.0', 0);
```
Comment (Chris-Petty) [inline src/utilities/compareVersions.js]: Missed some example of
testHelper('2.0.0', '1.0.0', 1);

File: src/utilities/compareVersions.js (hunk 2 above)
Comment (Chris-Petty) [reply src/utilities/compareVersions.js]: This is the only comment I actually want addressed haha

File: src/utilities/compareVersions.js (hunk 1 above)
Comment (andreievg) [reply src/utilities/compareVersions.js]: Nice table, looking at it i just realised if someone there is no number after RC, Number('') will return zero, that should not be a problem or ?

File: src/utilities/compareVersions.js (hunk 2 above)
Comment (andreievg) [reply src/utilities/compareVersions.js]: fixed

File: src/utilities/compareVersions.js (hunk 4 above)
Comment (Chris-Petty) [reply src/utilities/compareVersions.js]: Why assume when you can test 😛

File: src/utilities/compareVersions.js (hunk 1 above)
Comment (Chris-Petty) [reply src/utilities/compareVersions.js]: Shouldn't be `x + 0 = x` after all. More test cases! haha

File: src/utilities/compareVersions.js (hunk 4 above)
Comment (andreievg) [reply src/utilities/compareVersions.js]: Sure testing is the key. To me it seemed like 1 + 2 = 2 + 1

**Pull request merged**